docs/sources/flow/concepts/clustering.md

```diff
@@ -0,0 +1,243 @@
+---
+title: Clustering
+weight: 500
+labels:
+  stage: beta
+---
+
+# Clustering (beta)
+
+Clustering enables Grafana Agent Flow to coordinate a fleet of agents working
+together for workload distribution and high availability. It helps create
+horizontally scalable deployments with minimal resource and operational
+overhead.
+
+To achieve this, Grafana Agent makes use of an eventually consistent model that
+assumes all participating Agents are interchangeable and converge on using the
+same configuration file.
+
+The behavior of a standalone, non-clustered agent is the same as if it was a
+single-node cluster.
+
+In comparison to [hashmod sharding][] as a scaling mechanism, clustering with
+target auto-distribution provides high availability without requiring multiple
+replicas for the same shard, and can dynamically reshard targets without
+changes to the configuration file.
```

Comment:
Hm, we haven't really ever talked about the hashmod sharding approach w/r/t Flwo mode, and we haven't ever linked to static mode docs from Flow mode either. 

Should we drop the comparison and only talk about the benefits of clustering? 

cc @clayton-cornell

Reply:
IMO I like the focus on what we are offering and ignoring hashmod/scraping service mode.

Reply:
Ok, leaving the comparisons out for now.